packages/react/src/components/AccountSettings/DeleteUser/DeleteUser.tsx

```diff
@@ -0,0 +1,102 @@
+import React from 'react';
+
+import { deleteUser, translate, getLogger } from '@aws-amplify/ui';
+
+import { useAuth } from '../../../internal';
+import { Flex } from '../../../primitives';
+import {
+  DefaultWarning,
+  DefaultError,
+  DefaultSubmitButton,
+} from './defaultComponents';
+import { DeleteUserProps, DeleteUserState } from './types';
+
+const logger = getLogger('Auth');
+
+function DeleteUser({
+  onSuccess,
+  onError,
+  handleDelete,
+}: DeleteUserProps): JSX.Element | null {
+  const [state, setState] = React.useState<DeleteUserState>('IDLE');
+  const [errorMessage, setErrorMessage] = React.useState<string>(null);
+
+  // translations
+  const deleteAccountText = translate('Delete Account');
+
+  const { user, isLoading } = useAuth();
+
+  const startConfirmation = (event: React.MouseEvent<HTMLButtonElement>) => {
+    event.preventDefault();
+    setState('CONFIRMING');
+  };
+
+  const runDeleteUser = React.useCallback(async () => {
+    setState('DELETING');
+    try {
+      if (handleDelete) {
+        /*
+         * run custom delete handler, if provided. We pass `user` so that
+         * developer can do whichever cleanup with the user object they wish.
+         */
+        await handleDelete(user);
+      } else {
+        // else, run default deleteUser function.
+        await deleteUser();
+      }
+      setState('DONE');
+      onSuccess?.();
+    } catch (e) {
+      const error = e as Error;
+      setState('IDLE');
+      setErrorMessage(error.message);
+      onError?.(error);
+    }
+  }, [handleDelete, onError, onSuccess, user]);
+
+  // called when end user cancels account deletion confirmation
+  const handleCancel = React.useCallback(() => {
+    setState('IDLE');
+  }, []);
+
+  const handleConfirmDelete = React.useCallback(() => {
+    runDeleteUser();
+  }, [runDeleteUser]);
```

Comment:
Nit: **pretty** sure this could be simplified to 
```suggestion
const handleConfirmDelete = React.useCallback(runDeleteUser, [runDeleteUser]);